### DISCUSSION TITLE
Waiting on a completed Signal

### DISCUSSION THREAD

**Discussion by whitingjr** [Hyperfoil/qDup#196] [General]
The signal and wait-for commands are very useful for co-ordination.
 I need to check what the expected behaviour is for a wait-for.

 In a scenario there are two scripts. One calls `signal` the other calls `wait-for` with the same name.

 What happens when the script that calls signal command completes before the the second script (wait-for) starts.
 Does qDup keep a history of signals that were already emitted allowing the second script to progress beyond the wait-for or does it just hang forever ?

**Comment by willr3** [comment]
the `wait-for` should pass execution to the next command if the `signal` already occurred. We test to make sure a `signal` will even work if it occurs in a previous stage (e.g. setup)
https://github.com/Hyperfoil/qDup/blob/master/src/test/java/io/hyperfoil/tools/qdup/RunTest.java#L1010
